funding.py

```diff
@@ -184,149 +238,160 @@
+        ubi_population = population - child_population
+        ubi = revenue / ubi_population
+        spmu['total_ubi'] = ubi * (spmu.numper - spmu.child)
     
-    if 'unemp' in benefits:
-        funding += (target_persons.incunemp * target_persons.asecwt).sum()
-        target_persons.new_spm_resources -= target_persons.incunemp
-        
-    if 'eitc' in benefits:
-        funding += (target_persons.eitcred * target_persons.asecwt).sum()
-        target_persons.new_spm_resources -= target_persons.spmeitc
-        
-    if 'ctc' in benefits:
-        funding += (target_persons.ctc * target_persons.asecwt).sum()
-        target_persons.new_spm_resources -= target_persons.spmctc
-        
-    if 'snap' in benefits:
-        funding += (target_persons.snap_pp * target_persons.asecwt).sum()
-        target_persons.new_spm_resources -= target_persons.snap_pp
-        
-    if 'energy' in benefits:
-        funding += (target_persons.energy_pp * target_persons.asecwt).sum()
-        target_persons.new_spm_resources -= target_persons.energy_pp
+    if ('children' not in exclude) and ('non_citizens' in exclude):
+        ubi_population = population - non_citizen_population
+        ubi = revenue / ubi_population
+        spmu['total_ubi'] = ubi * (spmu.numper - spmu.non_citizen)
     
-    if 'income_taxes' in taxes:
-        funding -= (target_persons.fedtaxac * target_persons.asecwt).sum()
-        target_persons.new_spm_resources += target_persons.fedtaxac
+    if ('children' in exclude) and ('non_citizens' in exclude):
+        ubi_population = population - non_citizen_population - child_population
+        ubi = revenue / ubi_population
+        spmu['total_ubi'] = ubi * (spmu.numper - spmu.child - spmu.non_citizen)
     
-    if 'fica' in taxes:
-        funding -= (target_persons.fica * target_persons.asecwt).sum()
-        target_persons.new_spm_resources += target_persons.fica
+    else: 
+        ubi_population = population
+        ubi = revenue / ubi_population
+        spmu['total_ubi'] = ubi * spmu.numper
     
-    if ('income_taxes' in taxes) & ('ctc' in benefits):
-        funding -= (target_persons.ctc * target_persons.asecwt).sum()
-        target_persons.new_spm_resources += target_persons.spmctc
+    # Calculate 
+    spmu.new_resources += spmu.total_ubi
+    spmu['new_resources_per_person'] = (spmu.new_resources /
+                                                  spmu.numper)
     
-    if ('income_taxes' in taxes) & ('eitc' in benefits):
-        funding -= (target_persons.eitcred * target_persons.asecwt).sum()
-        target_persons.new_spm_resources += target_persons.spmeitc
+    # Calculate poverty gap
+    spmu['new_poverty_gap'] = np.where(spmu.new_resources < spmu.spmthresh, 
+                               spmu.spmthresh - spmu.new_resources, 0)
+    poverty_gap = mdf.weighted_sum(spmu, 'new_poverty_gap', 'spmwt')
+    poverty_gap_change = ((poverty_gap - original_poverty_gap) /
```

Comment:
Use a `chg` function like in the flat tax piece, see https://github.com/UBICenter/blog/blob/master/jb/20201230/data/generate_data.py#L63